dali/python/nvidia/dali/ops.py

```diff
@@ -13,18 +13,56 @@
 # limitations under the License.
 
 #pylint: disable=no-member
+from __future__ import division
 import sys
 import copy
 from itertools import count
 import threading
 from nvidia.dali import backend as b
-from nvidia.dali.edge import EdgeReference
 from nvidia.dali.types import _type_name_convert_to_string, _type_convert_value, DALIDataType, \
                               CUDAStream
 from nvidia.dali.pipeline import Pipeline
 from future.utils import with_metaclass
 import nvidia.dali.libpython_function_plugin
 
+
+class _EdgeReference(object):
+    def __init__(self, name, device="cpu", source=None):
+        self.name = name
+        self.device = device
+        self.source = source
+
+    # Note: Regardless of whether we want the cpu or gpu version
+    # of a tensor, we keep the source argument the same so that
+    # the pipeline can backtrack through the user-defined graph
+    def gpu(self):
+        return _EdgeReference(self.name, "gpu", self.source)
+
+    def __add__(self, other):
+        return _arithm_op("add", self, other)
+    def __radd__(self, other):
+        return _arithm_op("add", other, self)
+
+    def __sub__(self, other):
+        return _arithm_op("sub", self, other)
+    def __rsub__(self, other):
+        return _arithm_op("sub", other, self)
+
+    def __mul__(self, other):
+        return _arithm_op("mul", self, other)
+    def __rmul__(self, other):
+        return _arithm_op("mul", other, self)
+
+    def __truediv__(self, other):
```

Comment:
What are the semantics here?
Are we promoting division results to floating point?

Reply:
From now on, yes